hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java

```diff
@@ -116,11 +134,10 @@ public static HoodieWriteConfig createMetadataWriteConfig(
             // Below config is only used if isLogCompactionEnabled is set.
             .withLogCompactionBlocksThreshold(writeConfig.getMetadataLogCompactBlocksThreshold())
             .build())
-        .withParallelism(parallelism, parallelism)
-        .withDeleteParallelism(parallelism)
-        .withRollbackParallelism(parallelism)
-        .withFinalizeWriteParallelism(parallelism)
-        .withAllowMultiWriteOnSameInstant(true)
+        .withStorageConfig(HoodieStorageConfig.newBuilder().hfileMaxFileSize(maxHFileSizeBytes)
+            .logFileMaxSize(maxLogFileSizeBytes).logFileDataBlockMaxSize(maxLogBlockSizeBytes).build())
+        .withRollbackParallelism(defaultParallelism)
+        .withFinalizeWriteParallelism(defaultParallelism)
```

Comment:
did you remove .withAllowMultiWriteOnSameInstant(true) intentionally ?

Reply:
Yes. This was required because the previous code for commits would overwrite the same instant if it already exists. With the already commited rollback PR, if we get a commit on MDT with same timestamp as previous applied deltacommit then we will first rollback the previously applied deltacommit on MDT and then commit the new change. Hence, multi-write-on-same-instant is never possible in MDT.